<?xml version="1.0" encoding="utf-8"?>
<launch>
  <arg name="computer_name" value="usl" />
  <param name = "/use_sim_time" value="false"/>
  
  <include file="$(find calibration)/launch/rotate_os_cloud.launch" />

  <node  pkg="chkrbrd_plane_motion_detection" name="chkrbrd_plane_motion_detection_node_os_left" type="chkrbrd_plane_motion_detection_node" output="screen">
	<remap from="/lidar_in_topic" to="/os1_cloud_node/points/rotated"/>
	<remap from="/camera_in_topic" to="/nerian_stereo/left_image"/>
  	<param name = "x_min" type= "double" value= "0.75"/>
  	<param name = "x_max" type= "double" value= "6"/>
  	<param name = "y_min" type= "double" value= "-1"/>
  	<param name = "y_max" type= "double" value= "1"/>
  	<param name = "z_min" type= "double" value= "-1"/>
  	<param name = "z_max" type= "double" value= "3"/>
  	<param name = "min_pts" type= "int" value= "800"/>
  	<param name = "ransac_threshold" type= "double" value= "0.02"/>
        <param name="cam_config_file_path" value="$(find cam_lidar_calib)/config/nerian_passthru_faccal_left.yaml"/>
	<param name = "dx" type="double" value="0.0615"/>
	<param name = "dy" type="double" value="0.0615"/>

	<param name = "checkerboard_rows" type = "int" value = "9" />
	<param name = "checkerboard_cols" type = "int" value = "6" />
        <param name = "buffer_size" value="10"/>
        <param name="data_folder_name" value="$(find chkrbrd_plane_motion_detection)/data/left_os"/>
  </node>

  <node  pkg="chkrbrd_plane_motion_detection" name="chkrbrd_plane_motion_detection_node_os_right" type="chkrbrd_plane_motion_detection_node" output="screen">
	<remap from="/lidar_in_topic" to="/os1_cloud_node/points/rotated"/>
	<remap from="/camera_in_topic" to="/nerian_stereo/right_image"/>
  	<param name = "x_min" type= "double" value= "0.75"/>
  	<param name = "x_max" type= "double" value= "6"/>
  	<param name = "y_min" type= "double" value= "-1"/>
  	<param name = "y_max" type= "double" value= "1"/>
  	<param name = "z_min" type= "double" value= "-1"/>
  	<param name = "z_max" type= "double" value= "3"/>
  	<param name = "min_pts" type= "int" value= "800"/>
  	<param name = "ransac_threshold" type= "double" value= "0.02"/>
        <param name="cam_config_file_path" value="$(find cam_lidar_calib)/config/nerian_passthru_faccal_right.yaml"/>
	<param name = "dx" type="double" value="0.0615"/>
	<param name = "dy" type="double" value="0.0615"/>

	<param name = "checkerboard_rows" type = "int" value = "9" />
	<param name = "checkerboard_cols" type = "int" value = "6" />
        <param name = "buffer_size" value="10"/>
        <param name="data_folder_name" value="$(find chkrbrd_plane_motion_detection)/data/right_os"/>
  </node>
</launch>
